app/services/puppet_fact_parser.rb

```diff
@@ -55,6 +60,7 @@ def environment
   end
 
   def architecture
+    return @architecture if @architecture.present?
```

Comment:
Your right ill take care of it.
On Jun 3, 2015 12:34 AM, "Daniel Lobato García" <notifications@github.com>
wrote:

> In app/services/puppet_fact_parser.rb
> <https://github.com/theforeman/foreman/pull/2417#discussion_r31573867>:
>
> > @@ -55,6 +60,7 @@ def environment
> >    end
> >
> >    def architecture
> > +    return @architecture if @architecture.present?
>
> If I'm not wrong you could use attr_accessor here to avoid these guard
> clauses.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/theforeman/foreman/pull/2417/files#r31573867>.
>